hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -208,6 +203,26 @@ private void submitCooperativeTasklets(
         Arrays.stream(cooperativeThreadPool).forEach(LockSupport::unpark);
     }
 
+    private void awaitAll(List<? extends Future<?>> futures) {
+        Exception firstFailure = null;
+        int failureCount = 0;
+        for (Future<?> future : futures) {
+            try {
+                future.get();
+            } catch (InterruptedException | ExecutionException e) {
+                logger.severe("Tasklet initialization failed", e);
+                firstFailure = firstFailure != null ? firstFailure : peel(e);
+                failureCount++;
+            }
+        }
+        if (firstFailure != null) {
+            throw new JetException(String.format(
+                    "%,d of %,d tasklets failed to initialize. One of the failures is attached as the cause.",
```

Comment:
It's better to add the `cause` to the message. Some tools hide the stack trace and causes, then you won't see nothing, maybe mancenter does that too.
```java
"%,d of %,d tasklets failed to initialize. One of the failures: " + firstFailure,
```